maltoolbox/attackgraph/node.py

```diff
@@ -181,6 +183,101 @@ def is_available_defense(self) -> bool:
             'suppress' not in self.tags and \
             self.defense_status != 1.0
 
+    def ttc_sample(self) -> float:
+        """Sample a value from ttc distribution for a node
+
+        TTC Distributions in MAL:
+        https://mal-lang.org/mal-langspec/apidocs/org.mal_lang.langspec/org/mal_lang/langspec/ttc/TtcDistribution.html
+        """
+
+        def sample(exponential: float, bernoulli=1.0):
+            """
+            Generate a random sample for the given distributions.
+            If bernoulli distribution is not given, the sample will
+            simply be from an exponential.
+
+            If the Bernoulli trial fails (0), return infinity (impossible).
+            If the Bernoulli trial succeeds (1), return sample from
+            exponential distribution.
+            """
+
+            # If bernoulli is set to 1, the sample is just exponential
+            if np.random.choice([0, 1], p=[1 - bernoulli, bernoulli]):
+                return np.random.exponential(scale=1 / exponential)
+            return math.inf
```

Comment:
I think this is too convoluted. Why not use `np.random.exponential` directly in all cases in the lines below, except the bernoulli one where you do just random.choice? The way it is now, it seems like you are doing Bernoulli * Exponential, when the TTC in the mal spec does may not use the Bernoulli distribution at all and you have to default to the Bernoulli part returning 1 for such cases. While the result is the same, it is unnecessarily complex.

Reply:
After I wrote this, I realized that most of the distributions here are the codenamed ones, eg hard and uncertain which do use the Bernoulli distribution in their formula. But still others, like the pure exponential, do not and I still think that this could be simplified a bit. 

Maybe it merits an online call to discuss some aspects of this.